iphone/Classes/KrollContext.m

```diff
@@ -1529,3 +1536,59 @@ - (void)dealloc
 }
 
 @end
+
+@interface JSTimerManager : NSObject
+
++ (void)initializeInContext:(JSContext *)context;
+
++ (NSMutableDictionary<NSNumber *, NSTimer *> *)timers;
+
+@end
+
+@implementation JSTimerManager
+
++ (void)initializeInContext:(JSContext *)context {
+  int (^setInterval)(void) = ^() {
+    return [self setIntervalInContext:context withArguments:JSContext.currentArguments shouldRepeat:YES];
+  };
+  context[@"setInterval"] = setInterval;
+  
+  int (^setTimeout)(void) = ^() {
+    return [self setIntervalInContext:context withArguments:JSContext.currentArguments shouldRepeat:NO];
+  };
+  context[@"setTimeout"] = setTimeout;
+  
+  void (^clearInterval)(JSValue *) = ^(JSValue *value) {
+    return [self clearIntervalInContext:JSContext.currentContext withIdentifier:value.toInt32];
+  };
+  context[@"clearInterval"] = clearInterval;
+  context[@"clearTimeout"] = clearInterval;
+}
+
++ (NSMutableDictionary<NSNumber *, NSTimer *> *)timers {
+  static NSMutableDictionary *timers = nil;
+  if (timers == nil) {
+    timers = [NSMutableDictionary new];
+  }
+  return timers;
+}
+
++ (int)setIntervalInContext:(JSContext *)context withArguments:(NSArray<JSValue *> *)arguments shouldRepeat:(BOOL)repeats {
+  JSValue *callbackFunction = [[arguments objectAtIndex:0] retain];
+  double interval = [[arguments objectAtIndex:1] toDouble] / 1000;
+  NSNumber *timerIdentifier = [NSNumber numberWithUnsignedInteger:[[[[NSUUID alloc] init] autorelease] hash]];
+  NSTimer *timer = [NSTimer scheduledTimerWithTimeInterval:interval repeats:repeats block:^(NSTimer * _Nonnull timer) {
+    [callbackFunction callWithArguments:nil];
+    [callbackFunction release];
```

Comment:
@vijaysingh-axway not sure if it's a good idea to directly release the callback again here. Any thoughts?

Reply:
@janvennemann 
1. In case of setTimeout we should remove timer from self.timers array.
2. Is there any reason you have retained in callbackFunction. There is neither need of retain in callbackFunction nor release. Above code is crashing for me.
Test case -
`setTimeout(function(e){
      Ti.API.info('timeout called');
}, 2000);

setTimeout(function(e){
      Ti.API.info('timeout called second time');
}, 4000);

setInterval(function(e){
      Ti.API.info('time interval 1 called');
}, 3000);

setInterval(function(e){
      Ti.API.info('time interval 2 called');
}, 4000);`